metrics.go

```diff
@@ -461,22 +461,28 @@ func protoToMetricPoint(value interface{}) (*monitoringpb.TypedValue, error) {
 					Count:                 dv.Count,
 					Mean:                  mean,
 					SumOfSquaredDeviation: dv.SumOfSquaredDeviation,
-					BucketCounts:          bucketCounts(dv.Buckets),
 				},
 			}
 
+			insertZeroBound := false
 			if bopts := dv.BucketOptions; bopts != nil && bopts.Type != nil {
 				bexp, ok := bopts.Type.(*metricspb.DistributionValue_BucketOptions_Explicit_)
 				if ok && bexp != nil && bexp.Explicit != nil {
+					insertZeroBound = shouldInsertZeroBound(bexp.Explicit.Bounds...)
 					mv.DistributionValue.BucketOptions = &distributionpb.Distribution_BucketOptions{
 						Options: &distributionpb.Distribution_BucketOptions_ExplicitBuckets{
 							ExplicitBuckets: &distributionpb.Distribution_BucketOptions_Explicit{
-								Bounds: bexp.Explicit.Bounds[:],
+								// The first bucket bound should be 0.0 because the Metrics first bucket is
+								// [0, first_bound) but Stackdriver monitoring bucket bounds begin with -infinity
+								// (first bucket is (-infinity, 0))
+								Bounds: addZeroBound(insertZeroBound, bexp.Explicit.Bounds...),
```

Comment:
Optional nit: for better readability consider
```go
if insertZeroBound {
  addZeroBound(bexp.Explicit.Bounds...)
}
```

Reply:
@songy23 are you ok with the renaming the method or you want me to make the change as shown above?

Reply:
Renaming the method sounds good.